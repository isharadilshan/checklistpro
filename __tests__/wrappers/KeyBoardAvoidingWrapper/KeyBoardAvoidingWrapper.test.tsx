import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import {View, Text} from 'react-native';

import KeyBoardAvoidingWrapper from '../../../src/components/wrappers/KeyBoardAvoidingWrapper';

configure({adapter: new Adapter()});

const props = {
  children: (
    <View>
      <Text>Test Text</Text>
    </View>
  ),
  noPaddings: true,
};

describe('when rendering', () => {
  const wrapper = shallow(<KeyBoardAvoidingWrapper {...props} />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
