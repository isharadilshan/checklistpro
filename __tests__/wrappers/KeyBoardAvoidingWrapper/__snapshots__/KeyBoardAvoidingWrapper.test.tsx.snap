// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<ImageBackground
  resizeMode="cover"
  source={
    {
      "testUri": "../../../src/assets/background/abstract.png",
    }
  }
  style={
    {
      "backgroundColor": "black",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <KeyboardAvoidingView
    behavior="padding"
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 16,
        "width": "100%",
      }
    }
  >
    <View>
      <Text>
        Test Text
      </Text>
    </View>
  </KeyboardAvoidingView>
</ImageBackground>
`;
